File: kotlin/bc-crypto/src/test/kotlin/com/blockchaincommons/bccrypto/ArgonTest.kt
package com.blockchaincommons.bccrypto

import kotlin.test.Test
import kotlin.test.assertContentEquals
import kotlin.test.assertEquals
import kotlin.test.assertFalse

class ArgonTest {

    @Test
    fun testArgon2idBasic() {
        val pass = "password".toByteArray()
        val salt = "example salt".toByteArray()
        val output = argon2id(pass, salt, 32)
        assertEquals(32, output.size)
        val output2 = argon2id(pass, salt, 32)
        assertContentEquals(output, output2)
    }

    @Test
    fun testArgon2idDifferentSalt() {
        val pass = "password".toByteArray()
        val salt1 = "example salt".toByteArray()
        val salt2 = "example salt2".toByteArray()
        val out1 = argon2id(pass, salt1, 32)
        val out2 = argon2id(pass, salt2, 32)
        assertFalse(out1.contentEquals(out2))
    }
}

File: go/bccrypto/argon_test.go
package bccrypto

import "testing"

func TestArgon2IDBasic(t *testing.T) {
	pass := []byte("password")
	salt := []byte("example salt")
	out1 := Argon2ID(pass, salt, 32)
	if len(out1) != 32 {
		t.Fatalf("len = %d, want 32", len(out1))
	}
	out2 := Argon2ID(pass, salt, 32)
	if string(out1) != string(out2) {
		t.Fatalf("argon2id not deterministic")
	}
}

func TestArgon2IDDifferentSalt(t *testing.T) {
	pass := []byte("password")
	out1 := Argon2ID(pass, []byte("example salt"), 32)
	out2 := Argon2ID(pass, []byte("example salt2"), 32)
	if string(out1) == string(out2) {
		t.Fatalf("argon2id outputs unexpectedly equal")
	}
}
